test/sorting.jl

```diff
@@ -18,16 +18,23 @@ end
 @test sum(randperm(6)) == 21
 @test nthperm([0,1,2],3) == [1,0,2]
 
-@test searchsorted([1, 1, 2, 2, 3, 3], 0) == 1:0
-@test searchsorted([1, 1, 2, 2, 3, 3], 1) == 1:2
-@test searchsorted([1, 1, 2, 2, 3, 3], 2) == 3:4
-@test searchsorted([1, 1, 2, 2, 3, 3], 4) == 7:6
-@test searchsorted([1.0, 1, 2, 2, 3, 3], 2.5) == 5:4
-
-@test searchsorted([1:10], 1, by=(x -> x >= 5)) == 1:4
-@test searchsorted([1:10], 10, by=(x -> x >= 5)) == 5:10
-@test searchsorted([1:5, 1:5, 1:5], 1, 6, 10, Base.Order.Forward) == 6:6
-@test searchsorted(ones(15), 1, 6, 10, Base.Order.Forward) == 6:10
+for R in [Int, Uint8, Float64], T in [Int, Uint8, Float64]
```

Comment:
Seems like the list of types could be much longer? Why don't test `BigInt`, `BigFloat`, `UInt64`, `UInt128`....?